core/modules/util/common.h

```diff
@@ -93,6 +94,81 @@ std::ostream& printList(std::ostream &os, char const* label, C const& c) {
     return os;
 }
 
+template <typename C>
```

Comment:
These prettyChar* functions are fine, but they look prone to off-by-one errors. Would you add a test case that calls each of them once? I think a well-intentioned future dev might try to "clean up" things and introduce an error and spend way too long debugging it.